test/integration/test_expid.py

```diff
@@ -260,3 +279,337 @@ def test_copy_expid_no(autosubmit: Autosubmit) -> None:
+    mocker.patch("shutil.rmtree", side_effect=FileNotFoundError)
+    mocker.patch("os.remove", side_effect=FileNotFoundError)
+    basic_config = as_exp.as_conf.basic_config
+    experiment_path = Path(f"{basic_config.LOCAL_ROOT_DIR}/{_EXPID}")
+    structure_db_path = Path(f"{basic_config.STRUCTURES_DIR}/structure_{_EXPID}.db")
+    job_data_db_path = Path(f"{basic_config.JOBDATA_DIR}/job_data_{_EXPID}")
+    if all("tmp" not in path for path in [str(experiment_path), str(structure_db_path), str(job_data_db_path)]):
+        raise AutosubmitCritical("tmp not in path")
+    mocker.patch("autosubmit.autosubmit.delete_experiment", side_effect=FileNotFoundError)
+    err_message = autosubmit._perform_deletion(experiment_path, structure_db_path, job_data_db_path, _EXPID)
+    assert all(x in err_message for x in
+               ["Cannot delete experiment entry", "Cannot delete directory", "Cannot delete structure",
+                "Cannot delete job_data"])
+
+
+@pytest.mark.parametrize(
+    "project_type,expid",
+    [
+        ('git', 'o001'),
+        ('git', 'a001'),
+        ('git', 't001'),
+        ('git', 'e001'),
+    ]
+)
+def test_remote_repo_operational(project_type: str, expid: str, tmp_path: Path, autosubmit_exp: Callable,
+                                 autosubmit, mocker: MockerFixture) -> None:
+    """
+    Tests the check_unpushed_changed function from AutosubmitGit.
+
+    Ensures no operational test with unpushed changes in their Git repository is run.
+    """
+    as_exp = autosubmit_exp(expid, create_as_exp(tmp_path))
+    run_dir = Path(as_exp.as_conf.basic_config.LOCAL_ROOT_DIR)
+    temp_path = run_dir / expid / "conf" / f"expdef_{expid}.yml"
+
+    with open(temp_path, 'r') as f:
+        yaml = YAML(typ='rt')
+        data = yaml.load(f)
+    data["PROJECT"]["PROJECT_TYPE"] = project_type
+    with open(temp_path, 'w') as f:
+        yaml.dump(data, f)
+
+    mocker.patch('subprocess.check_output', return_value=b'M\n')
+    if expid[0] != 'o':
+        AutosubmitGit.check_unpushed_changes(expid)
+    else:
+        with pytest.raises(AutosubmitCritical):
+            AutosubmitGit.check_unpushed_changes(expid)
+
+
+@pytest.mark.parametrize(
+    "git_repo, expected",
+    [
+        ("https://github.com/user/repo.git", True),  # valid GH link
+        ("file:///home/user/project", True),  # valid file link
+        ("not-a-repo-link", False),  # invalid
+        ("git@github.com:user/repo.git", True),  # SSH format
+        ("user@gitserver.com:user/repo.git", True),  # SSH format
+        ("http://bitbucket.org/user/repo.git", True),  # valid git host
+        ("ftp://invalid/protocol/repo.git", False),  # invalid protocol
+        ("", False),  # empty string
+        ("file://", False),  # incomplete file URL
+        ("https://github.com/user/repo", False),  # missing .git
+    ]
+)
+def test_valid_git_repo_check(git_repo: str, expected: str) -> None:
+    assert AutosubmitGit.is_git_repo(git_repo) == expected
```

Comment:
Hmmm, this looks like a test for `autosubmit_git`. Maybe move it to a new test for that file.